### PR TITLE
feat(kafka): support JSON-syntax for message key

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## Description

Added supporting JSON-syntax in message key for Outbound and Inbound connectors;
Added e2e tests

## Related issues

https://github.com/camunda/connectors/issues/1456

### CODE REVIEW

File: connectors-e2e-test/connectors-e2e-test-kafka/src/test/java/io/camunda/connector/e2e/BaseKafkaTest.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  @Autowired ZeebeClient zeebeClient;
+
+  @MockBean ProcessDefinitionSearch processDefinitionSearch;
+
+  @Autowired InboundConnectorManager inboundManager;
+
+  @Autowired CamundaOperateClient camundaOperateClient;
+
+  static KafkaContainer kafkaContainer;
```
Comment (sbuettner) [inline connectors-e2e-test/connectors-e2e-test-kafka/src/test/java/io/camunda/connector/e2e/BaseKafkaTest.java:68]: Great to see that we are using TestContainers now in our e2e tests. 🎉

File: connectors-e2e-test/connectors-e2e-test-kafka/src/test/java/io/camunda/connector/e2e/BaseKafkaTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import org.junit.jupiter.api.io.TempDir;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class BaseKafkaTest {
+
+  static final String TEST_GROUP_ID = "test-group";
+  static final String HEADER_KEY_VALUE = "{\"headerKey\":\"headerValue\"}";
```
Comment (sbuettner) [inline connectors-e2e-test/connectors-e2e-test-kafka/src/test/java/io/camunda/connector/e2e/BaseKafkaTest.java:47]: I am wondering why the json is escaped here?

File: connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaPropertyTransformer.java [hunk 3]
```diff
@@ -93,22 +99,36 @@ public static KafkaInboundMessage convertConsumerRecordToKafkaInboundMessage(
         kafkaInboundMessage.setHeaders(headerMap);
       }
     }
+  }
 
+  private static Object parseKey(
+      ConsumerRecord<Object, Object> consumerRecord, ObjectReader objectReader) {
+    try {
+      String key = (String) consumerRecord.key();
+      var json = StringEscapeUtils.unescapeJson(key);
```
Comment (sbuettner) [inline connectors/kafka/src/main/java/io/camunda/connector/kafka/inbound/KafkaPropertyTransformer.java:108]: Curious, why do we need to unescape json here?

**Review by igpetrov — APPROVED** [review]
Nicely done

**Pull request merged**